GVFS/GVFS.FunctionalTests/Tests/FastFetchTests.cs

```diff
@@ -185,9 +200,10 @@ public void CanFetchAndCheckoutBranchIntoEmptyGitRepo()
 
             this.fastFetchRepoRoot.ShouldBeADirectory(FileSystemRunner.DefaultRunner)
                 .WithDeepStructure(FileSystemRunner.DefaultRunner, this.fastFetchControlRoot);
-        }
-
-        [TestCase]
+        }
+
+        [TestCase]
+        [Category(Categories.MacTODO.M4)]

```

Comment:
Why are some of the tests failing on Mac?

Reply:
The two tests that are failing are due to the way we create the index in FastFetch. I'll address these in a follow up PR to make this one smaller. 

In the tests, we compare the index created by FastFetch with one that 'git status' creates. We compare them by 'git ls-files --debug' which outputs in this format. 

Settings.StyleCop
  ctime: 1540245129:777055100
  mtime: 1540245129:777055100
  **dev: 0        ino: 0**
  uid: 0        gid: 0
  size: 10738   flags: 0
nuget.config
  ctime: 1540245129:777055100
  mtime: 1541723805:620569300
  **dev: 0        ino: 0**
  uid: 0        gid: 0
  size: 617     flags: 0

The differences between the two indexes ends up being that dev/ino are 0 in FastFetch's index while they have real random values in the git created index.

I believe the fix is to just ignore dev/ino when comparing the ls-files output but I want to dig a little deeper before sweeping this under the rug.

More context:
These are unix internals that track the /dev device that contains the file and a unique inode number for the file. These don't mean anything on Windows, so Git for Windows writes these out as 0s (experimentally, still need to dig through the source to prove this). As far as I can tell, Git doesn't use these values for anything.

https://github.com/git/git/blob/master/Documentation/technical/index-format.txt
```
        // We can't accurated fill times and length in realtime, so **we block write the zeroes and probably save time.**
        private static readonly byte[] EntryHeader = new byte[] 
        {
            0, 0, 0, 0,
            0, 0, 0, 0, // ctime
            0, 0, 0, 0,
            0, 0, 0, 0, // mtime
            0, 0, 0, 0, // stat(2) dev
            0, 0, 0, 0, // stat(2) ino
            0, 0, 0x81, 0xA4, // filemode (0x81A4 in little endian)
            0, 0, 0, 0, // stat(2) uid
            0, 0, 0, 0, // stat(2) gid
            0, 0, 0, 0  // file length
        };
```

Reply:
Could you file a follow up issue for this?  It would be good to confirm if it's OK that FastFetch leaves these fields as 0 on Mac.

CC: @kewillford who might be more familiar with how git uses these fields

Reply:
The index is a cache of what was on disk that last time status ran.  In most cases a status call will just update these values the next time it runs and stats the file.  There are places that the fields are used like [match_stat_data](https://github.com/Microsoft/git/blob/43ac10287170d00704cf69aca4702c28ffd2a03d/read-cache.c#L150) but that is mainly to know if the index entry needs to be refreshed.

Reply:
Thanks guys! I went ahead and made a change here to remove the lines starting with dev: and uid: when doing the index comparison in the tests. This behaves as expected.